### PR TITLE
Fix field errors

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

- Created a component to render consistent form help/error/success messages
- Fixes error message on samples file upload and create new sample

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Create a sample with an existing sample name, error message should be displayed properly
2. On a file, click upload a file, an click `Upload file` without actually selecting a file.  Error message should be displayed properly

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/components/viral/form/help_text_component.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      }.freeze
+
+      def initialize(state: :default, **system_arguments)
+        @state = state
+        @arguments = system_arguments
+        @icon = ICON_MAPPINGS.key?(@state) ? ICON_MAPPINGS[@state] : ICON_MAPPINGS[:default]
+      end
+
+      def system_arguments
+        @arguments.tap do |args|
```
Comment (ksierks) [inline app/components/viral/form/help_text_component.rb:28]: This is a little embarrassing, but I've never heard about or seen the method tap before, lol.

**Review by ksierks — APPROVED** [review]
The form validation messages looks nicer now, thanks.

File: app/components/viral/form/help_text_component.rb (hunk 1 above)
Comment (joshsadam) [reply app/components/viral/form/help_text_component.rb:28]: Yeah, it's pretty handy.  I think I first saw it in the Polaris code base.  Also a small right up on it here: https://medium.com/aviabird/ruby-tap-that-method-90c8a801fd6a

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**